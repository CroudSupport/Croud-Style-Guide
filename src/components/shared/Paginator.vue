<template>
    <h1>{{ test }}</h1>
</template>

<script>
    /**
     * This is the croud paginator.
     * @example ./croud-paginator.md
     */
    export default {
        name: 'croud-paginator',

        props: {
            /**
             * The color for the button.
             */
            test: {
                type: String,
                default: 'test',
            },
        },
    }
</script>
