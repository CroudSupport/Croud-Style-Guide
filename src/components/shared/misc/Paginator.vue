<template>
    <div class="ui small pagination menu" v-if="pagination.total_pages > 1 && pagination.total_pages <= 5">
        <span v-for="page in pages">
              <a class="item" :class="{active: current === page}" @click="current = page">{{ page }}</a>
        </span>
    </div>

    <div class="ui small pagination borderless menu" v-else-if="pagination.total_pages > 1">
        <a :class="startClass" @click="current = 1">First Page</a>
        <a :class="startClass" @click="current -= 1">Prev</a>
        <a class="item">Page {{current}} of {{pagination.total_pages}}</a>
        <a :class="endClass" class="item" @click="current += 1">Next</a>
        <a :class="endClass" @click="current = pagination.total_pages">Last Page</a>
    </div>
</template>

<script>
    /**
    * @example ./croud-paginator.md
    */
    export default {
        name: 'croud-paginator',
        model: {
            prop: 'currentPage',
        },
        props: {
            /**
            *
            */
            pagination: {
                required: true,
                type: Object,
            },
            /**
            * Page Currently being displayed
            */
            currentPage: {
                type: Number,
                required: true,
            },
        },

        computed: {
            current: {
                get() {
                    return this.currentPage
                },
                set(page) {
                    this.$emit('set-current', page)
                    this.$emit('input', page)
                },
            },

            pages() {
                /* eslint-disable prefer-spread */
                return Array.apply(null, { length: this.pagination.total_pages }).map((v, k) => k + 1)
            },

            startClass() {
                return this.current > 1 ? 'item' : 'item disabled'
            },

            endClass() {
                return this.current < this.pagination.total_pages ? 'item' : 'item disabled'
            },
        },
    }
</script>
