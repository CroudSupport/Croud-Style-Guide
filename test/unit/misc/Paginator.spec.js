import Vue from 'vue'
import Paginator from '../../../src/components/shared/misc/Paginator'

const Constructor = Vue.extend(Paginator)

const vm = new Constructor({
    propsData: {
        pagination: {
            total_pages: 12,
        },
        currentPage: 5,
    },
}).$mount()

describe('Paginator', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })
})
