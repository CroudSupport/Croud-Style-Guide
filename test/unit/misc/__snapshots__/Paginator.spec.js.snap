// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator should match the snapshot 1`] = `
<div
  class="ui small pagination borderless menu"
>
  <a
    class="item"
  >
    First Page
  </a>
  <a
    class="item"
  >
    Prev
  </a>
  <a
    class="item"
  >
    Page 5 of 12
  </a>
  <a
    class="item item"
  >
    Next
  </a>
  <a
    class="item"
  >
    Last Page
  </a>
</div>
`;
